inject-java/src/main/java/io/micronaut/annotation/processing/visitor/AbstractJavaElement.java

```diff
@@ -211,30 +211,23 @@ public String toString() {
                 return mirrorToClassElement(bound, visitorContext, genericsInfo);
             } else {
 
-                ClassElement classElement = mirrorToClassElement(upperBound, visitorContext, genericsInfo);
-                if (classElement != null) {
-                    return classElement;
-                } else {
-                    return mirrorToClassElement(tv.getLowerBound(), visitorContext, genericsInfo);
-                }
+                return mirrorToClassElement(upperBound, visitorContext, genericsInfo);
```

Comment:
This doesn't seem correct

Reply:
`mirrorToClassElement` never returns null.